src/main/java/uk/gov/justice/digital/delius/service/CustodyService.java

```diff
@@ -273,6 +287,17 @@ public Custody offenderReleased(final String nomsNumber, final OffenderReleasedN
             });
     }
 
+    private void addRelease(OffenderReleasedNotification releasedNotification, Long eventId, String crn) {
+        var context = getContext(CONTEXT);
+        var releaseMapping = context.getReleaseTypeMapping();
+        var deliusReleaseType = ofNullable(releaseMapping.getReasonToReleaseType().get(releasedNotification.getReason())).orElseThrow(
+            () -> new BadRequestException("Release Type mapping from reason does not exist for: " + releasedNotification.getReason())
+        );
+
+        val newRelease = new NewRelease(releasedNotification.getReleaseDate(), deliusReleaseType, releasedNotification.getNomsPrisonInstitutionCode());
```

Comment:
Just an option, but you could make use of the `@Builder` here and ditch the constructor if you wanted.
```suggestion
        val newRelease = NewRelease.builder()
                .releaseType(deliusReleaseType)
                .actualReleaseDate(releasedNotification.getReleaseDate())
                .institution(releasedNotification.getNomsPrisonInstitutionCode())
                .build();
```